### PR TITLE
Make undef'd counts zero by default.

### LINKED ISSUE #15409: 'make update' failure
Doing a make update:
```
/usr/bin/perl ./util/mknum.pl --version 3.0.0-beta1-dev --no-warnings \
                --ordinals ./util/libcrypto.num \
                --symhacks ./include/openssl/symhacks.h \
                include/internal/asn1.h include/internal/dso.h include/internal/err.h 
                ..list of header files trimmed ...  include/openssl/x509v3err.h
./util/libcrypto.num: No new symbols added
Use of uninitialized value in subtraction (-) at ./util/mknum.pl line 142.
```

### PR DESCRIPTION
**Pull request by richsalz**
Fixes #15409

### CODE REVIEW

File: util/mknum.pl [hunk 1]
```diff
@@ -138,8 +138,9 @@
     }
     if ($dropped) {
         print STDERR "${ordinals_file}: Dropped $dropped new symbols\n";
+        $stats{unassigned} = 0 unless defined $stats{unassigned};
+        $unassigned = $stats{unassigned} - $dropped;
```
Comment (t8m) [inline util/mknum.pl:142]: Shouldn't those 2 lines be outside the if() block?

**Review by t8m — CHANGES_REQUESTED** [review]

File: util/mknum.pl (hunk 1 above)
Comment (richsalz) [reply util/mknum.pl:142]: > Shouldn't those 2 lines be outside the if() block?

`$stats{unassigned}` isn't used except here, and why subtract zero? :)  So no.

Comment (t8m) [reply util/mknum.pl:142]: Huh? But `$unassigned` is used down below and you're happily leaving it without any value assigned.

File: util/mknum.pl [hunk 2]
```diff
@@ -138,8 +138,9 @@
...
     }
-    $unassigned = $stats{unassigned} - $dropped;
```
Comment (levitte) [inline util/mknum.pl:142]: That looks a bit clumsy.  How about this?

``` perl
    if ($dropped) {
        print STDERR "${ordinals_file}: Dropped $dropped new symbols\n";
    }
    $unassigned = $stats{unassigned} // 0 - $dropped;
```

File: util/mknum.pl (hunk 1 above)
Comment (richsalz) [reply util/mknum.pl]: > Huh? But `$unassigned` is used down below and you're happily leaving it without any value assigned.

replacing one 'unassigned' bug with another. :)  fixed, thanks. udpated commit pushed.

File: util/mknum.pl (hunk 2 above)
Comment (richsalz) [reply util/mknum.pl]: I find my change to be more clear, even if it is less perlish magic.

Comment (levitte) [reply util/mknum.pl]: Uhmm, `//` is a perfectly normal perl operator, there's really nothing magic about it.

(I suppose you know this bashism:  `${foo:-something}` ...  that's about as magic that it is)

**Review by t8m — APPROVED** [review]

File: util/mknum.pl (hunk 2 above)
Comment (richsalz) [reply util/mknum.pl]: https://www.in-ulm.de/~mascheck/bourne/v7/  I know `${foo?bar}` etc.  Never used bash much.  I prefer "rc"

Comment (kaduk) [reply util/mknum.pl]: Not bashism.  POSIX sh.

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the contribution.

**Pull request closed without merging**